src/useSafeState.ts

```diff
@@ -0,0 +1,25 @@
+import { Dispatch, SetStateAction, useCallback, useState } from 'react';
+import { useIsMounted } from './useIsMounted';
+
+/**
+ * Like `useState` but its state setter is guarded against sets on unmounted component.
+ *
+ * @param initialState
+ */
+export function useSafeState<S>(initialState: S | (() => S)): [S, Dispatch<SetStateAction<S>>];
+export function useSafeState<S = undefined>(): [
+  S | undefined,
+  Dispatch<SetStateAction<S | undefined>>
+];
+export function useSafeState<S>(initialState?: S | (() => S)): [S, Dispatch<SetStateAction<S>>] {
+  const [state, _setState] = useState(initialState);
```

Comment:
Why `_setState`  instead of just `setState`?

Reply:
Actually i wanted to make local `const` but then decided to straight export callback %)